staging/src/k8s.io/kubectl/pkg/proxy/proxy_server.go

```diff
@@ -173,8 +173,8 @@ func makeUpgradeTransport(config *rest.Config, keepalive time.Duration) (proxy.U
 
 // NewServer creates and installs a new Server.
 // 'filter', if non-nil, protects requests to the api only.
-func NewServer(filebase string, apiProxyPrefix string, staticPrefix string, filter *FilterServer, cfg *rest.Config, keepalive time.Duration) (*Server, error) {
-	proxyHandler, err := NewProxyHandler(apiProxyPrefix, filter, cfg, keepalive)
+func NewServer(filebase string, apiProxyPrefix string, staticPrefix string, filter *FilterServer, cfg *rest.Config, keepalive time.Duration, appendLocationPath bool) (*Server, error) {
```

Comment:
@lavalamp How bad is change the NewServer function signature to add a new option at this stage?

Reply:
I don't know what compatibility guarantees SIG CLI offers on kubectl code-- I wouldn't expect any but I don't know.

Reply:
@lavalamp Thanks, let me put into the thread @soltysh. Maciej, could you please review this PR and share your thoughts please?

Reply:
Yeah, kubectl-wise I'm ok with extending this method signature.